### PR TITLE
Add very high level description to std.format

### PR DESCRIPTION
**Pull request by kuettler**

### CODE REVIEW

File: std/format.d [hunk 1]
```diff
@@ -3,12 +3,44 @@
...
+)
+
+   Please see the documentation of function $(D $(LREF formattedWrite)) for a
+   description of the _format string.
+
+   Two functions have been added for convenience:
+
+$(BOOKTABLE ,
+    $(TR $(TD $(D $(LREF _format)))
+        $(TD Returns a newly allocated string with the formatting result.
```
Comment (Hackerpilot) [inline std/format.d]: It may be worth changing this to "GC-allocated" from "newly allocated" for the sake of clarity.

**Comment by Hackerpilot** [comment]
:white_check_mark:

**Comment by kuettler** [comment]
Thanks.

File: std/format.d [hunk 2]
```diff
@@ -3,12 +3,44 @@
...
+    ))
+)
+
+   Please see the documentation of function $(D $(LREF formattedWrite)) for a
+   description of the _format string.
+
+   Two functions have been added for convenience:
+
+$(BOOKTABLE ,
+    $(TR $(TD $(D $(LREF _format)))
```
Comment (JakobOvrum) [inline std/format.d]: `format` and `sformat` are in std.string

Comment (kuettler) [inline std/format.d]: Actually, `format` and `sformat` are publicly imported in std.string. I guess both functions moved to std.format at some point in time.

Comment (JakobOvrum) [inline std/format.d]: Indeed, nevermind!

**Comment by JakobOvrum** [comment]
LGTM

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by kuettler** [comment]
So I pushed a comment change to trigger the autotester again. It failed in std.file on Windows, certainly unrelated to this.

**Comment by kuettler** [comment]
Thanks a lot! :) There is also #2997 that goes along the same lines for string. (Just an improvement, honestly.)

**Pull request merged**